docs/book/programs/xdp.md

```diff
@@ -1,3 +1,516 @@
 # XDP
 
-This page is a work in progress, please feel free to open a Pull Request!
+!!! example "Source Code"
+
+    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-drop)
+
+
+## What is XDP in eBPF?
+XDP (eXpress Data Path) is a type of eBPF program that attaches to the network interface.
+It enables filtering, manipulation and refirection of network packets
+as soon as they are received from the network driver,
+even before they enter the Linux kernel networking stack, resulting in low latency and high throughput.
+
+The idea behind XDP is to add an early hook in the `RX` path of the kernel,
+and let a user supplied eBPF program decide the fate of the packet.
+The hook is placed in the NIC driver just after the interrupt processing,
+and before any memory allocation needed by the network stack itself.
+
+The XDP program is allowed to edit the packet data and,
+after the XDP program returns, an action code determines what to do with the packet:
+
+* `XDP_PASS`: let the packet continue through the network stack
+* `XDP_DROP`: silently drop the packet
+* `XDP_ABORTED`: drop the packet with trace point exception
+* `XDP_TX`: bounce the packet back to the same NIC it arrived on
+* `XDP_REDIRECT`: redirect the packet to another NIC or user space socket via the
+[`AF_XDP`](https://www.kernel.org/doc/html/latest/networking/af_xdp.html) address family
+
+## AF_XDP
+Along with XDP, a new address familiy entered in the Linux kernel, starting at 4.18.
+`AF_XDP`, formerly known as `AF_PACKETv4` (which was never included in the mainline kernel),
+is a raw socket optimized for high performance packet processing and
+allows zero-copy between kernel and applications.
+As the socket can be used for both receiving and transmitting,
+it supports high performance network applications purely in user-space.
+
+If you want a more extensive explanation about `AF_XDP`,
+you can find it in the [kernel documentation](https://www.kernel.org/doc/html/latest/networking/af_xdp.html).
+
+## XDP Operation Modes
```

Comment:
I'm personally not a big fan of using the [title case](https://en.wikipedia.org/wiki/Title_case) (capitalizing all nouns, regardless of whether they're proper nouns or common nouns). As far as I remember, @alessandrod doesn't like it either.

The reasons being:

- I Find it Quite Difficult to Read. :stuck_out_tongue: 
- In all seriousness, I like the opposite style - sentence case capitalization - because it highlights the important words. When reading technical documentation, I prefer the names of specific technologies or projects (XDP, Linux etc.) to stand out from the rest of the title.
- Title case is not really standardized, there are multiple styles and rules. I find sentence case capitalization much more obvious, since everyone is using it on daily basis.
- In Aya book, even though there are some places using title case, it's not being used consistently. It's also not being used consistently here. :sweat_smile: It would be easier to convert all existing title cases to sentence cases - I would be happy to see an another PR doing that.

Therefore, 

```suggestion
## XDP operation modes

```